### PR TITLE
script-loader: Improve API, spread any props on to the script

### PR DESCRIPTION
**Pull request by thetarnav**
Add `onBeforeAppend` callback option. Rename onload and onerror to onLoad and onError. Do not return the remove script function - should be done with disposing the owner. React to src changes with createRenderEffect.

### CODE REVIEW

**Comment by atk** [comment]
I don't like extra events, I'd very much prefer if we could just add normal props as we would for a component. See my latest branch.

**Comment by thetarnav** [comment]
Ok, I like this, it's much cleaner. and removes the need for other events as well.
Should I merge yours into this branch? there are still other changes here

**Comment by atk** [comment]
Yes, please.

**Review by atk — APPROVED** [review]

File: packages/script-loader/src/index.ts [hunk 1]
```diff
@@ -1,43 +1,45 @@
...
 /**
  * Creates a convenient script loader utility
  *
- * @param string URL or source of the script to load.
- * @param type Type value to put in the script attribute.
- * @param function Callback to trigger onLoad.
- * @param function callback on error.
- * @returns
+ * @param props The props to spread to the script element.
+ * The `src` prop is required and will be used to set the `src` or `textContent` attribute.
```
Comment (atk) [inline packages/script-loader/src/index.ts:15]: ... And can also be a signal's getter.

**Review by atk — APPROVED** [review]

**Pull request merged**